sdk/common/src/main/java/io/opentelemetry/sdk/internal/DaemonThreadFactory.java

```diff
@@ -20,14 +21,20 @@ public final class DaemonThreadFactory implements ThreadFactory {
   private final String namePrefix;
   private final AtomicInteger counter = new AtomicInteger();
   private final ThreadFactory delegate = Executors.defaultThreadFactory();
+  private final boolean propagateContext;
 
   public DaemonThreadFactory(String namePrefix) {
+    this(namePrefix, /* propagateContext= */ false);
+  }
+
+  public DaemonThreadFactory(String namePrefix, boolean propagateContext) {
```

Comment:
#nit: can we rename this parameter to `propagateContextForTesting`, and maybe add a comment indicating its usage for testing? I want to add some small guardrails / description of this parameter, which would be rather confusing to come across without the context of this PR.

Reply:
Sure, I've made the changes